### ISSUE TITLE
Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported

### ISSUE THREAD

**Issue by bhz020418** [flutter/flutter#120404]
So after I ran:
flutter build ipa
I got the output like this in visual studio code:

```

💪 Building with sound null safety 💪

Archiving com.ehealth.EMSdailysurvey...
Automatically signing iOS for device deployment using specified development team in Xcode project: 54GG7M63R4
Running pod install...                                             786ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         5.4s
Xcode archive done.                                         39.1s
Failed to build iOS app
Error output from Xcode build:
↳
    error: using bridging headers with module interfaces is unsupported
    ** ARCHIVE FAILED **


Xcode's output:
↳
    Writing result bundle at path:
        /var/folders/9p/bw8m2_ns5kx4hppr37mzz1zr0000gn/T/flutter_tools.9yTkx8/flutter_ios_build_temp_dir6Hgplm/temporary_xcresult_bundle

    error: Using bridging headers with module interfaces is unsupported (in target 'Runner' from project 'Runner')
    Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure
    note: Building targets in dependency order
    warning: Run script build phase '[CP-User] Config codegen' will be run during every build because it does not specify any outputs. To address this
    warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the
    script phase. (in target 'flutter_config' from project 'Pods')
    warning: no rule to process file '/Users/wendasai/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_config-2.0.0/ios/Classes/BuildDotenvConfig.rb'
    of type 'text.script.ruby' for architecture 'arm64' (in target 'flutter_config' from project 'Pods')
    warning: no rule to process file '/Users/wendasai/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_config-2.0.0/ios/Classes/BuildXCConfig.rb' of
    type 'text.script.ruby' for architecture 'arm64' (in target 'flutter_config' from project 'Pods')
    warning: no rule to process file '/Users/wendasai/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_config-2.0.0/ios/Classes/ReadDotEnv.rb' of type
    'text.script.ruby' for architecture 'arm64' (in target 'flutter_config' from project 'Pods')
    note: Run script build phase 'Run Script' will be run during every build because the option to run the script phase "Based on dependency analysis" is
    unchecked. (in target 'Runner' from project 'Runner')
    note: Run script build phase 'Thin Binary' will be run during every build because the option to run the script phase "Based on dependency analysis" is
    unchecked. (in target 'Runner' from project 'Runner')

    Result bundle written to path:
        /var/folders/9p/bw8m2_ns5kx4hppr37mzz1zr0000gn/T/flutter_tools.9yTkx8/flutter_ios_build_temp_dir6Hgplm/temporary_xcresult_bundle

Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported

Uncategorized (Xcode): Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure


Encountered error while archiving for device.

```

The application can be built properly using IOS simulator. I really start to get very confuse about why it does not allow me to build the ipa file. Any one have any idea why this is happening? I am trying to compile the ipa for a IOS program I wrote but I have been stuck on this bug for a week... I do not know how to solve this. I googled online and it seems there are no similar solutions... Can anyone help me please! Thank you in advance!

Regards

**Comment by darshankawar** [comment]
@bhz020418 
Please check [this](https://stackoverflow.com/a/60658142) and see if it helps in your case.

**Comment by farhanvk8** [comment]
I have this same issue, setting Build Libraries for Distribution to NO doesn't resolve this.

**Comment by bhz020418** [comment]
> @bhz020418 Please check [this](https://stackoverflow.com/a/60658142) and see if it helps in your case.

Hello @darshankawar:

I tried the method listed in this post but is still not resolved, I am using Xcode version 14.2. So I decided to send the post on GitHub because I really failed to understand individually about why this is happening... 

But still thank you for sending me this post, if you have any other idea about how to solve this issue please let me know. Thank you in advance and wish you a nice day! (^-^)

Regards
bhz020418

**Comment by bhz020418** [comment]
> I have this same issue, setting Build Libraries for Distribution to NO doesn't resolve this.
> 
> `Xcode archive done. 236.0s Failed to build iOS app Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported
> 
> Uncategorized (Xcode): Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure
> 
> Encountered error while archiving for device.
> 
> Build failed :| Failed to build for iOS`
> 
> Flutter version 3.7.3 Xcode Version 14.2

Hello @farhanvk8:

Same here, I tried the method listed above by @darshankawar few days ago and it is not solved, I am using flutter version 3.7.0 and Xcode version 14.2. I am using mac computer with 2020 M1 chip. I did not upgrade to 3.7.3 for my flutter yet because after I updated the flutter to 3.7.3, the application wrote cannot simulate properly on the IOS simulator so I changed it back to 3.7.0.

If you have any new update about how to solve this issue please let me know. Thank you so much and wish you a nice day. I will also work on trying other methods and if I solved the issue I will let you know at once.

Regards
bhz020418

**Comment by darshankawar** [comment]
@bhz020418 
Please provide your `flutter doctor -v`. Also provide the entire verbose log by running `flutter build ipa -v`. The log will be huge so you can put that in a text file and attach the file here.

**Comment by bhz020418** [comment]
> @bhz020418 Please provide your `flutter doctor -v`. Also provide the entire verbose log by running `flutter build ipa -v`. The log will be huge so you can put that in a text file and attach the file here.

Hello @darshankawar:

Here is my flutter doctor -v and my flutter build ipa -v. If there is any other things that you need me to provide please let me know. 

Looking forward to your response, thank you so much and wish you a nice day.

Regards
bhz020418
[flutter doctor -v.txt](https://github.com/flutter/flutter/files/10726920/flutter.doctor.-v.txt)
[flutter build ipa -v.txt](https://github.com/flutter/flutter/files/10726922/flutter.build.ipa.-v.txt)

**Comment by bhz020418** [comment]
> @bhz020418 Please provide your `flutter doctor -v`. Also provide the entire verbose log by running `flutter build ipa -v`. The log will be huge so you can put that in a text file and attach the file here.

And @darshankawar here is the error I can see after I entered flutter build ipa. Hope this can also help. (^-^)
<img width="1440" alt="recent problem" src="https://user-images.githubusercontent.com/111715018/218593127-26af1c83-2505-40c4-a2f4-23810873296c.png">

Regards
bhz020418

**Comment by darshankawar** [comment]
@bhz020418 Can you also provide your `podfile` ?
Do you by any chance have checked "For install builds only" in the Xcode build phase ?

**Comment by bhz020418** [comment]
> @bhz020418 Can you also provide your `podfile` ? Do you by any chance have checked "For install builds only" in the Xcode build phase ?

Hello @darshankawar:

Sure! Here is my Podfile.lock and my Podfile. I did not check "for install builds only".

Regards
bhz020418 


[Podfile.txt](https://github.com/flutter/flutter/files/10732987/Podfile.txt)
[Podfile_dot_lock.txt](https://github.com/flutter/flutter/files/10732984/Podfile_dot_lock.txt)
<img width="1440" alt="Image for 'install build only'" src="https://user-images.githubusercontent.com/111715018/218757909-56cdc936-75d0-4294-98ce-b178d91f8150.png">

**Comment by darshankawar** [comment]
I see that in your podfile, you have `ios 12` platform support and target set to 12 as well:

`config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'`

I am not sure but this maybe restricting other pods or plugins you might be using which may not be targeting to iOS 12 yet, so can you try to comment or remove above line from podfile and try again to see if it helps ?

**Comment by bhz020418** [comment]
> I see that in your podfile, you have `ios 12` platform support and target set to 12 as well:
> 
> `config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'`
> 
> I am not sure but this maybe restricting other pods or plugins you might be using which may not be targeting to iOS 12 yet, so can you try to comment or remove above line from podfile and try again to see if it helps ?

Hello @darshankawar:

I changed commented the line as shown in the figure attached below. But when I entered flutter build ipa it still output the same error... The IOS simulator is still running properly. 

Looking forward to your response. 

Regards
bhz020418
<img width="1440" alt="Screen Shot 2023-02-15 at 12 17 20 AM" src="https://user-images.githubusercontent.com/111715018/218940507-e44c8b2c-08e6-4224-8419-58dba1b5b902.png">

**Comment by darshankawar** [comment]
@bhz020418 Thanks for the update and your patience on this while we figure this out.
From your `podfile.lock`, I see the the pods you are using are not on latest versions. Can you try to delete the `podfile.lock` file or update the pods you are using to latest versions available and run your app again ?

**Comment by bhz020418** [comment]
> 

Hello @darshankawar:

I deleted the podfile.lock and then ran flutter build ipa again. The error I am encountering with is as shown in figure 1 (after I ran flutter build ipa, the podfile.lock is back again). Moreover, I also tried to run it in xcode and the output I got is shown in figure 2 - figure 3 or figure 4 - figure 13 (sometimes I got the errors and warnings as shown in figure 2 - figure 3(1 error and 7 warnings), sometimes I got the errors and warnings as shown in figure 4 - figure 13 (1 error and 137 warnings)). Also, If I am using the simulator on my computer, the code can be run properly (as shown in figure 14). I just really do not quite understand why my xcode is stopping me from building the ipa package, which is making feeling so frustrating. (T-T)

Also I do not know how to update the pods I am using to latest versions available. If possible, can you tell me how to do it? 

Looking forward to your response, Thank you so much and wish you a nice day.

Regards
bhz020418

<img width="1440" alt="figure 1" src="https://user-images.githubusercontent.com/111715018/219423824-6a8623f4-239c-4993-ac94-33958b1c1bba.png">
<img width="1440" alt="figure 2" src="https://user-images.githubusercontent.com/111715018/219423900-73cb0539-7021-455e-b35a-4b51ea8957a6.png">
<img width="1440" alt="figure 3" src="https://user-images.githubusercontent.com/111715018/219423905-e7ba0bcd-54be-4914-860b-2ffb8e118cd8.png">
<img width="1440" alt="figure 4" src="https://user-images.githubusercontent.com/111715018/219423908-af7c4a1a-ca35-4232-a637-7300155627d3.png">
<img width="1440" alt="figure 5" src="https://user-images.githubusercontent.com/111715018/219423912-f2135451-f9ba-44f5-bc13-fa849fb1b96b.png">
<img width="1440" alt="figure 6" src="https://user-images.githubusercontent.com/111715018/219423914-2795255b-94de-4c04-9326-b083c67ed3d4.png">
<img width="1440" alt="figure 7" src="https://user-images.githubusercontent.com/111715018/219423916-5018d7bb-a3bb-44f8-b307-d0eac758a15f.png">
<img width="1440" alt="figure 8" src="https://user-images.githubusercontent.com/111715018/219423917-7647464a-b22f-4aee-9e08-2f5eb89d8334.png">
<img width="1440" alt="figure 9" src="https://user-images.githubusercontent.com/111715018/219423922-17b7b482-2be0-416b-9197-307f54f9f762.png">
<img width="1440" alt="figure 10" src="https://user-images.githubusercontent.com/111715018/219423924-eb5016c8-a4a7-4c12-8f95-cb9f7bab3e29.png">
<img width="1440" alt="figure 11" src="https://user-images.githubusercontent.com/111715018/219423926-caeffbcb-5338-40d8-be29-70aaa33c5c3a.png">
<img width="1440" alt="figure 12" src="https://user-images.githubusercontent.com/111715018/219423927-ab2537a7-0191-4064-9db3-0958946e47ee.png">
<img width="1440" alt="figure 13" src="https://user-images.githubusercontent.com/111715018/219423930-8bfaf043-7b94-4ecc-9add-915e5992dc94.png">
<img width="1440" alt="figure 14" src="https://user-images.githubusercontent.com/111715018/219424343-413f6bc2-25f4-4a17-91cd-ceb0c0da370e.png">

**Comment by darshankawar** [comment]
@bhz020418 
Please avoid sharing screenshots as they don't help much for searching for errors mentioned in it. Please provide the error messages in properly formatted text.

Also, if possible, please provide a sample repo that we can clone and run it at our end to analyze further.

**Comment by bhz020418** [comment]
> @bhz020418 Please avoid sharing screenshots as they don't help much for searching for errors mentioned in it. Please provide the error messages in properly formatted text.
> 
> Also, if possible, please provide a sample repo that we can clone and run it at our end to analyze further.

Hello @darshankawar:

Sure! I asked my group members and we reached to the consensus that we can provide the sample repo. But before that we need to change some parts in our programming relating to the customer data, which will take some time to do the adjustment. So sorry for cannot providing it right now and so sorry for previously sending the screenshots because we were thinking perhaps that will help solving the problem...

When we finished adjusting the part relating to the user's data, we will reply back with the sample repo as soon as possible. Thank you so much for helping us all along the road and wish you the best. 

Regards
bhz020418
